src/go/rpk/pkg/cli/transform/logs.go

```diff
@@ -0,0 +1,523 @@
+/*
+* Copyright 2024 Redpanda Data, Inc.
+*
+* Use of this software is governed by the Business Source License
+* included in the file licenses/BSL.md
+*
+* As of the Change Date specified in that file, in accordance with
+* the Business Source License, use of this software will be governed
+* by the Apache License, Version 2.0
+ */
+
+package transform
+
+import (
+	"container/list"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"os/signal"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.uber.org/zap"
+)
+
+var (
+	minTime = time.Unix(0, 0).UTC()
+	maxTime = time.Unix(math.MaxInt64, math.MaxInt64).UTC()
+)
+
+func newLogsCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var follow bool
+	var head, tail int
+	var since, until timequery
+	since.time = minTime
+	until.time = maxTime
+	cmd := &cobra.Command{
+		Use:     "logs NAME",
+		Aliases: []string{"log"},
+		Short:   "View logs for a transform",
+		Long: `View logs for a transform.
+
+Data transform's STDOUT and STDERR are captured during runtime and written to a topic.
+This command allows viewing logs for a single transform over a period of time and printing them
+to STDOUT. The logs can be printed in various formats.
+
+The --follow flag allows for streaming logs continuously, by default only emitted logs are displayed.
+
+FILTERING
+
+The --head and --tail flags are mutually exclusive and allow for limiting the number of log entries
+from the beginning and end of the range, respectively.
+
+The --since and --until flags allow for limiting the logs to a period of time. 
```

Comment:
Can --since and --until be used at the same time?

Reply:
Yes! Is there a good way to make that more clear?

Reply:
I don't really understand how that's supposed to work. Maybe an easy example?

Reply:
Does something like the follow make sense?

```
# Limit logs between noon and 1pm UTC on march 12
rpk transform logs --since=2024-03-12T12:00:00Z --until=2024-03-12T13:00:00Z
```

Reply:
that does it. Thanks @rockwotj